### PR TITLE
Project restructorization

### PR DESCRIPTION
**Pull request by Vodorok**
Modified project structure according to a widely used template.

Replaced the old dependency copy mechanism that copied the needed jars
under the plugin/lib folder, with a more osgi compliant solution, made
possible with the p2-maven-plugin.

Added Oxygen compatible target platform. Now this is the lowest
officially supported release of this plugin. At the same time the
required lowest Eclipse-PDE should be Eclipse Oxygen 1.

Added Integration tests as well. Swtbot is being used to conduct the gui
testing. There is a suite with a dummy cpp project imported. The master
project is copied into the Junit-Workspace before every run, and gets
discarded after the run. The test projects reside in the resources
folder.

Added first true Integration test case that tests the "Add CodeChecker
Nature" context menu item functionality.

### CODE REVIEW

File: bundles/org.codechecker.eclipse.plugin.plugin/.project [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>org.codechecker.eclipse.plugin.plugin</name>
```
Comment (gyorb) [inline bundles/org.codechecker.eclipse.plugin.plugin/.project]: Do we need the two plugins here in the naming? Can it be done to move the package one level up?

Comment (Vodorok) [reply bundles/org.codechecker.eclipse.plugin.plugin/.project]: Refactored the whole thing, now the project base name is org.codechecker.eclipse.

**Comment by Vodorok** [comment]
Solved the project import problem in the integration tests. No more major modifications planned.

**Comment by Vodorok** [comment]
Will fix the Checkstyle problems in the integration tests project.

File: features/org.codechecker.eclipse.feature/feature.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<feature
+      id="org.codechecker.eclipse.feature"
+      label="CodeChecker Eclipse Plugin Core"
+      version="1.0.0.qualifier"
+      provider-name="Ericsson">
+
+   <description url="http://www.org.codechecker.eclipse.com/description">
```
Comment (gyorb) [inline features/org.codechecker.eclipse.feature/feature.xml]: Shouldn't this be a link to the github project page?

File: features/org.codechecker.eclipse.feature/feature.xml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      [Enter Feature Description here.]
+   </description>
+
+   <copyright url="http://www.org.codechecker.eclipse.com/copyright">
+      [Enter Copyright Description here.]
+   </copyright>
+
+   <license url="http://www.org.codechecker.eclipse.com/license">
```
Comment (gyorb) [inline features/org.codechecker.eclipse.feature/feature.xml]: Shouldn't this be a link to the github project page?

File: features/org.codechecker.eclipse.feature/feature.xml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      id="org.codechecker.eclipse.feature"
+      label="CodeChecker Eclipse Plugin Core"
+      version="1.0.0.qualifier"
+      provider-name="Ericsson">
+
+   <description url="http://www.org.codechecker.eclipse.com/description">
+      [Enter Feature Description here.]
+   </description>
+
+   <copyright url="http://www.org.codechecker.eclipse.com/copyright">
```
Comment (gyorb) [inline features/org.codechecker.eclipse.feature/feature.xml]: Shouldn't this be a link to the license file on github?

File: readme.md [hunk 5]
```diff
@@ -8,20 +8,22 @@ This is a C/C++ code analysis plugin for Eclipse that shows bugs detected by the
 
 * Linux operating system
 * Recent [CodeChecker](https://github.com/Ericsson/codechecker)
-* [Eclipse](www.eclipse.org): the plugin is currently tested with Eclipse Neon, but any recent Eclipse version should work
+* The last supported Eclipse version is [Oxygen 1](https://www.eclipse.org/downloads/packages/release/oxygen/1), version 4.7.0.
```
Comment (gyorb) [inline readme.md]: I would rephrase this a bit.

`Every eclipse version newer than [Oxygen 1](https://www.eclipse.org/downloads/packages/release/oxygen/1) v4.7.0 is supported.`

File: bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/runtime/CodeCheckEnvironmentChecker.java [hunk 6]
```diff
@@ -162,9 +162,9 @@ public String getLogFileLocation() {
      * @return The constructed analyze command.
      */
     public String createAnalyzeCommmand(String buildLog){
-        return codeCheckerCommand + " analyze " + getConfigValue(ConfigTypes.CHECKER_LIST) +
+        return "\"" + codeCheckerCommand + "\"" + " analyze " + getConfigValue(ConfigTypes.CHECKER_LIST) +
```
Comment (gyorb) [inline bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/runtime/CodeCheckEnvironmentChecker.java]: Please do not mix a bugfix in the restructure PR. Please create a separate PR for that.

File: tests/org.codechecker.eclipse.rcp.it.tests/resources/cproject/src/cproject.c [hunk 7]
```diff
@@ -0,0 +1,17 @@
+/*
```
Comment (gyorb) [inline tests/org.codechecker.eclipse.rcp.it.tests/resources/cproject/src/cproject.c]: Can you create a separate PR adding this new test? If yes please split it up.
Adding a test later will be easier if it is visible if we see what are the changes required to add a test, right now it is mixed with the restructure changes.

File: .travis.yml [hunk 8]
```diff
@@ -1,14 +1,24 @@
...
 before_script:
-- cd ../CodeCheckerEclipsePlugin/eclipse-plugin/
-script: mvn package
+    - sh -e /etc/init.d/xvfb start
+    - sleep 10 # give xvfb some time to start
+
+script: mvn -f mavendeps/pom.xml p2:site && mvn clean verify
+
+after_failure:
+    - cat /home/travis/build/Ericsson/CodeCheckerEclipsePlugin/tests/org.codechecker.eclipse.plugin.rcp.it.tests/target/work/data/.metadata/.log
```
Comment (gyorb) [inline .travis.yml]: Is there and environment variable for the travis build dir? It would be better instead of hard coding the path.

File: tests/org.codechecker.eclipse.rcp.it.tests/resources/cproject/src/cproject.c (hunk 7 above)
Comment (gyorb) [reply tests/org.codechecker.eclipse.rcp.it.tests/resources/cproject/src/cproject.c]: The PR for the separate test can be found here #154

File: bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/runtime/CodeCheckEnvironmentChecker.java (hunk 6 above)
Comment (gyorb) [reply bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/runtime/CodeCheckEnvironmentChecker.java]: @Vodorok will you do a separate PR for these changes too?
